### DISCUSSION TITLE
High CPU usage when installing from private repository

### DISCUSSION THREAD

**Discussion by dejan9393** [verdaccio/verdaccio#2119] [Q&A] [answered]
Hello,

We've recently hit an issue where our private GitLab instance is getting rammed with 600% cpu usage when we do an npm install through our Verdaccio registry.

I was under the assumption that Verdaccio stores the packages locally on the filesystem, so it would only need to hit the GitLab repo for metadata on the package, not for downloading packages. Any ideas?

**Comment by dejan9393** [comment, marked as answer]
I believe this is an issue with the [verdaccio-gitlab](https://github.com/bufferoverflow/verdaccio-gitlab) library. I submitted a PR to work around this issue here: https://github.com/bufferoverflow/verdaccio-gitlab/pull/126

**Reply by juanpicado** [reply]
Ok, seems is tracked there, please feel free to post here whenever is fixed there :)
